cpp/tests/gis/point_in_polygon_test.cu

```diff
@@ -0,0 +1,168 @@
+        
+        // Polygon must be closed
+        size_t size_polygon = polygon_lats.size();
+        EXPECT_EQ( polygon_lats[0], polygon_lats[size_polygon - 1] ) << "TEST: Latitudes of polygon must be closed.";
+        EXPECT_EQ( polygon_lats[0], polygon_lats[size_polygon- 1] ) << "TEST: Longitudes of polygon must be closed.";
+
+        if(print)
+        {
+            std::cout << "\nSize of polygon: " << polygon_lats.size() << std::endl;
+            std::cout << "Number of points: " << point_lats.size() << std::endl;
+        }
+    }
+
+    // Aditional host function
+    T orientation(T p1_x, T p1_y, T p2_x, T p2_y, T p3_x, T p3_y)
+    {
+	    return ((p2_y - p1_y) * (p3_x - p2_x) - (p2_x - p1_x) * (p3_y - p2_y));
+    }
+
+    std::vector<int8_t> compute_reference_pip()
+    {   
+        size_t total_points = point_lats.size();
+        std::vector<int8_t> h_inside_polygon(total_points, -1);
+
+        for (size_t id_point = 0; id_point < total_points; ++id_point)
+        {
+            T point_lat = point_lats[id_point];
+            T point_lon = point_lons[id_point];
+            int count = 0;
+
+            for (size_t poly_idx = 0; poly_idx < polygon_lats.size() - 1; ++poly_idx)
+            {
+                if(polygon_lons[poly_idx] <= point_lon && point_lon < polygon_lons[poly_idx + 1])
+                {
+                    if (orientation(polygon_lons[poly_idx], polygon_lats[poly_idx], polygon_lons[poly_idx + 1], polygon_lats[poly_idx + 1], point_lon, point_lat) > 0)
+                    {
+                        count++;
+                    }
+                }
+                else if (point_lon <= polygon_lons[poly_idx] && polygon_lons[poly_idx + 1] < point_lon) 
+                {
+                    if (orientation(polygon_lons[poly_idx], polygon_lats[poly_idx], polygon_lons[poly_idx + 1], polygon_lats[poly_idx + 1], point_lon, point_lat) > 0)
+                    {
+                        count++;
+                    }
+                }
+            }
+
+            if ((count > 0) && (count % 2 == 0)) h_inside_polygon[id_point] = 1;
+		    else h_inside_polygon[id_point] = 0;
+        }
+
+        return h_inside_polygon;
+    }
+
+    std::vector<int8_t> compute_gdf_pip()
+    {
+        cudf::test::column_wrapper<T> polygon_lat_wrapp{polygon_lats};
+        cudf::test::column_wrapper<T> polygon_lon_wrapp{polygon_lons};
+        cudf::test::column_wrapper<T> point_lat_wrapp{point_lats};
+        cudf::test::column_wrapper<T> point_lon_wrapp{point_lons};
+
+        gdf_column_poly_lats = polygon_lat_wrapp.get();
+        gdf_column_poly_lons = polygon_lon_wrapp.get();
+        gdf_column_point_lats = point_lat_wrapp.get();
+        gdf_column_point_lons = point_lon_wrapp.get();
+
+        gdf_column* inside_polygon_column = cudf::gdf_point_in_polygon(gdf_column_poly_lats, gdf_column_poly_lons, gdf_column_point_lats, gdf_column_point_lons);
```

Comment:
I think it would be simpler to not have gdf_column_poly_lats, gdf_column_poly_lons, etc and intermediate variables.